pkg/workloads/cortex/lib/storage/s3.py

```diff
@@ -250,6 +250,10 @@ def get_json_external(self, s3_path, num_retries=0, retry_delay_sec=2):
             return None
         return json.loads(obj.decode("utf-8"))
 
-    def list_objects(self, s3_path):
-        bucket, prefix = self.deconstruct_s3_path(s3_path)
-        return self.s3.list_objects(Bucket=bucket, Prefix=prefix)
+    def list_objects(self, path):
+        bucket, prefix = self.deconstruct_s3_path(path)
+        return [
+            obj["Key"]
+            for obj in self.s3.list_objects(Bucket=bucket, Prefix=prefix)["Contents"]
```

Comment:
This should be [list_objects_v2](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects_v2)